addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/internal/state/MonitorThingState.java

```diff
@@ -0,0 +1,537 @@
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.zoneminder.ZoneMinderConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import name.eskildsen.zoneminder.api.telnet.ZoneMinderTriggerEvent;
+import name.eskildsen.zoneminder.common.ZoneMinderMonitorFunctionEnum;
+import name.eskildsen.zoneminder.common.ZoneMinderMonitorStatusEnum;
+import name.eskildsen.zoneminder.data.IMonitorDataGeneral;
+import name.eskildsen.zoneminder.data.IZoneMinderEventData;
+
+/**
+ * The {@link MonitorThingState} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Martin S. Eskildsen - Initial contribution
+ */
+public class MonitorThingState extends GenericThingState {
+
+    private Logger logger = LoggerFactory.getLogger(MonitorThingState.class);
+
+    private boolean isDirty;
+    private String logIdentifier = "";
+
+    private IZoneMinderEventData activeEvent = null;
+    private ZoneMinderTriggerEvent curTriggerEvent = null;
+
+    /*
+     * Used in recalculate
+     */
+    AtomicBoolean recordingFunction = new AtomicBoolean(false);
+    AtomicBoolean recordingDetailedState = new AtomicBoolean(false);
+    AtomicBoolean alarmedFunction = new AtomicBoolean(false);
+    AtomicBoolean alarmedDetailedState = new AtomicBoolean(false);
+
+    // Monitor properties
+
+    // Video Url
+    private State channelVideoUrl;
+
+    boolean bRecalculating = false;
+
+    /*
+     * public MonitorThingState(String id, ChannelStateChangeSubscriber subscriber) {
+     * super(subscriber);
+     * logIdentifier = id;
+     * initialize();
+     * }
+     */
+    protected void initialize() {
+        isDirty = true;
+    }
+
+    public MonitorThingState(ChannelStateChangeSubscriber subscriber) {
+        super(subscriber);
+
+    }
+
+    @Override
+    public GenericChannelState createChannelSubscription(ChannelUID channelUID) {
+        GenericChannelState channelState = null;
+        try {
+            switch (channelUID.getId()) {
+                case ZoneMinderConstants.CHANNEL_MONITOR_DETAILED_STATUS:
+                case ZoneMinderConstants.CHANNEL_MONITOR_EVENT_CAUSE:
+                case ZoneMinderConstants.CHANNEL_MONITOR_FUNCTION:
+                    channelState = createSubscriptionStringType(channelUID);
+                    break;
+
+                case ZoneMinderConstants.CHANNEL_MONITOR_STILL_IMAGE:
+                    channelState = createSubscriptionRawType(channelUID);
+                    break;
+                default:
+                    channelState = createSubscriptionOnOffType(channelUID);
+                    break;
+            }
+
+        } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.